### ISSUE TITLE
efibootmgr fails at e8ce9fe

### ISSUE THREAD

**Issue by asavah** [rhboot/efibootmgr#119]
at commit e8ce9fe
```
 efibootmgr  -v
--reconnect is supported only for driver entries.
```
And it fails to add a boot entry too.

I don't have a fully working system at the moment (due to this bug) to properly bisect this.

d9eb7f1 seems to work
```
./efibootmgr -v
BootCurrent: 0002
Timeout: 1 seconds
BootOrder: 0002,0001
Boot0001  Hard Drive    BBS(HD,,0x0)/VenHw(5ce8128b-2cec-40f0-8372-80640e3dc858,0200)..GO..NO..........S.a.m.s.u.n.g. .S.S.D. .8.6.0. .E.V.O. .2.5.0.G.B...................\.,.@.r.d.=.X.........A................................>..Gd-.;.A..MQ..L.3.S.9.Y.X.N.M.0.1.2.7.9.4.9. .B. . . . .......BO..NO..........F.o.r.c.e. .M.P.5.1.0...................\.,.@.r.d.=.X.........A..........................dy..1g1.....>..Gd-.;.A..MQ..L.1.8.4.2.8.2.0.5.0.0.0.1.2.7.6.9.6.1.1.C.......BO
      dp: 05 01 09 00 02 00 00 00 00 / 01 04 16 00 8b 12 e8 5c ec 2c f0 40 83 72 80 64 0e 3d c8 58 02 00 / 7f ff 04 00
    data: 00 00 47 4f 00 00 4e 4f d1 00 00 00 08 00 00 00 8b 00 53 00 61 00 6d 00 73 00 75 00 6e 00 67 00 20 00 53 00 53 00 44 00 20 00 38 00 36 00 30 00 20 00 45 00 56 00 4f 00 20 00 32 00 35 00 30 00 47 00 42 00 00 00 05 01 09 00 02 00 00 00 00 01 04 16 00 8b 12 e8 5c ec 2c f0 40 83 72 80 64 0e 3d c8 58 02 00 7f ff 04 00 02 01 0c 00 d0 41 03 0a 00 00 00 00 01 01 06 00 02 01 01 01 06 00 01 00 03 12 0a 00 00 00 ff ff 00 00 7f ff 04 00 01 04 3e 00 ef 47 64 2d c9 3b a0 41 ac 19 4d 51 d0 1b 4c e6 33 00 53 00 39 00 59 00 58 00 4e 00 4d 00 30 00 31 00 32 00 37 00 39 00 34 00 39 00 20 00 42 00 20 00 20 00 20 00 20 00 00 00 7f ff 04 00 00 00 42 4f 00 00 4e 4f bb 00 00 00 08 00 00 00 91 00 46 00 6f 00 72 00 63 00 65 00 20 00 4d 00 50 00 35 00 31 00 30 00 00 00 05 01 09 00 02 00 00 00 00 01 04 16 00 8b 12 e8 5c ec 2c f0 40 83 72 80 64 0e 3d c8 58 02 00 7f ff 04 00 02 01 0c 00 d0 41 03 0a 00 00 00 00 01 01 06 00 06 01 01 01 06 00 00 00 03 17 10 00 01 00 00 00 64 79 a7 12 31 67 31 43 7f ff 04 00 01 04 3e 00 ef 47 64 2d c9 3b a0 41 ac 19 4d 51 d0 1b 4c e6 31 00 38 00 34 00 32 00 38 00 32 00 30 00 35 00 30 00 30 00 30 00 31 00 32 00 37 00 36 00 39 00 36 00 31 00 31 00 43 00 00 00 7f ff 04 00 00 00 42 4f
Boot0002* UEFI: PXE IP4 Realtek PCIe GBE Family Controller      PciRoot(0x0)/Pci(0x1,0x2)/Pci(0x0,0x2)/Pci(0x4,0x0)/Pci(0x0,0x0)/MAC(00d8610edd17,0)/IPv4(0.0.0.00.0.0.0,0,0)..BO
      dp: 02 01 0c 00 d0 41 03 0a 00 00 00 00 / 01 01 06 00 02 01 / 01 01 06 00 02 00 / 01 01 06 00 00 04 / 01 01 06 00 00 00 / 03 0b 25 00 00 d8 61 0e dd 17 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 / 03 0c 1b 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 / 7f ff 04 00
    data: 00 00 42 4f
```

The board is "msi b450i gaming plus ac" ofc. set to boot in UEFI mode.
And ofcourse efivars are mounted rw.

**Comment by slytomcat** [comment]
I'ts wrong check of ops.reconnect at lines 1741-1742 and there is correct and more full check of it on lines 1776-1779. So lines 1741-1742 can be removed. I've done it in #121

**Comment by guido99** [comment]
I see this, too. Only way to see all of the boot entries is

     efibootmgr -b 0018 -F

where 0018 is my current default entry.

**Comment by asavah** [comment]
There are 2 PRs which fix this issue, the right one being #121 , but the mantainer seems to not care that master has been broken for a month.
ping @vathpela

**Comment by vathpela** [comment]
Should be fixed now; re-open if this is still broken.
